File: 02-frontend/react-library-app/src/models/Book.ts
class Book {
    id: number;
    title: string;
    author?: string;
    overview?: string;
    overviewDe?: string;
    totalCopies?: number;
    copiesInStock?: number;
    category?: string;
    image?: string;
    deleted?: boolean;

    constructor(id: number, title: string, author: string, overview: string, overviewDe: string,
        totalCopies: number, copiesInStock: number, category: string,
        image: string, deleted: boolean) {
        this.id = id;
        this.title = title;
        this.author = author;
        this.overview = overview;
        this.overviewDe = overviewDe;
        this.totalCopies = totalCopies;
        this.copiesInStock = copiesInStock;
        this.category = category;
        this.image = image;
        this.deleted = deleted;
    }
}

export default Book;

File: 01-backend/spring-boot-library/src/main/java/com/libranova/spring_boot_library/model/Book.java
package com.libranova.spring_boot_library.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "book")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Book
{
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    @Column(name = "title")
    private String title;

    @Column(name = "author")
    private String author;

    @Column(name = "overview")
    private String overview;

    @Column(name = "overview_de")
    private String overviewDe;

    @Column(name = "total_copies")
    private int totalCopies;

    @Column(name = "copies_in_stock")
    private int copiesInStock;

    @Column(name = "isbn")
    private String isbn;

    @Column(name = "year_published")
    private Integer yearPublished;

    @Column(name = "category")
    private String category;

    @Column(name = "image")
    private String image;

    @Column(name = "deleted")
    private boolean deleted = false;
}
